### ISSUE TITLE
Add a kubectl command to generate manifests

### ISSUE THREAD

**Issue by cgag** [kubernetes/kubernetes#34148]
Currently if I want to create a deployment, I either copy and paste from the docs and modify as needed.  It would be nice to be able to run something like `kubectl new deployment` and have
a template spit to stdout like:
 
```
apiVersion: extensions/v1beta1
kind: Deployment
metadata:
  name: my-deployment
spec:
  replicas: 1
  template:
    metadata:
      labels:
        app: my-app
    spec:
      containers:
      - name: myapp
        image: myimage:mytag
        ports:
        - containerPort: 80
```

I actually wouldn't particularly mind if it just dumped out one of the examples from the current docs, filled with nginx.

**Comment by AdoHe** [comment]
@kubernetes/sig-cli @kubernetes/kubectl

**Comment by soltysh** [comment]
There's either `kubectl run nginx --image=nginx --dry-run -o yaml` which will do that for you, and/or #33250 that should make this easier to edit before submitting. Isn't that sufficient?

**Comment by ymqytw** [comment]
They will be another PR that will let you edit after the generator and before submitting. This will come after #33250. These 2 PRs will address #18064 and should be helpful for you.

**Comment by cgag** [comment]
@soltysh That would work in this case, though I'll have to think more about other resources.  However, I think that command is very unintuitive .  I only recently learned that run and create support -o=yaml, which I wouldn't expect because their normal output is nothing like a summary of the resource created.   Nor does "create a resource, but not really" (--dry-run) really map to the thought "I want a manifest".

As a more minor complaint, that output contains output I like `creationTimestamp: null` that is less than ideal.

@soltysh + @ymqytw: I'm not sure edit helps, dry-run fits this use case more clearly, I don't just want to edit the config before shipping it to the cluster, I want it persisted on disk.   I think edit solving "I have a template, I want to fill in values before shipping", rather than "I want a fresh template".

**Comment by AdoHe** [comment]
I agree that users may want a fresh template sometimes.

**Comment by deads2k** [comment]
I think this fits nicely with the goals of `kubectl create`.  We've used this downstream to have `oc create deploymentconfig the-name --image=nginx` which produces the minimal object required.   We've started having these commands in kube as well, but we haven't created a similar one yet.

**Comment by deads2k** [comment]
@fabianofranz you'll probably get some mileage out of an upstream port here.  Remember the `kubectl create` guidelines for minimal flags (https://github.com/kubernetes/kubernetes/blob/master/docs/devel/kubectl-conventions.md#create-commands)

**Comment by soltysh** [comment]
With `kubectl create` you still need to pass `-o yaml` but I think that's perfectly reasonable. I don't see any justification for additional commands that will spit the object back at you by default.

**Comment by pwittrock** [comment]
Closing as dup of #25382
